src/adagrams.js

```diff
@@ -1,15 +1,121 @@
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
 export const drawLetters = () => {
-  // Implement this method for wave 1
+
+  let drawedLetters = [];
+  const numAllowedinHand = 10;
+  let allLetters = [];
+
+  for (const [key, value] of Object.entries(LETTER_POOL)){
+    for (let i = 0; i < value; i++){
+      allLetters.push(key);
+    };
+  };
+
+  while (drawedLetters.length < numAllowedinHand){
+    let randomNumber = Math.floor(Math.random() * allLetters.length);
+    drawedLetters.push(allLetters[randomNumber]);
+    allLetters.splice(randomNumber,1);
+  
+  };
+  
+  return drawedLetters;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  let lettersFreq = {};
+
+  for (const letter of input){
+      lettersFreq[letter] = (lettersFreq[letter] || 0) + 1;
+  };
+
+  let letterHandFreq = {};
+  for (const letter of lettersInHand){
+    letterHandFreq[letter] = (letterHandFreq[letter] || 0) + 1;
+  };
+
+  for (const letter of input){
+    if (!(letter in letterHandFreq) || lettersFreq[letter] > letterHandFreq[letter]){
+      return false;
+    };
+  };
+  return true;
 };
 
 export const scoreWord = (word) => {
+  const SCORECHART = {1:['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+             2:['D','G'],3:['B','C','M','P'],
+             4:['F','H','V','W','Y'],5:['K'],
+             8:['J','X'],10:['Q','Z']}
   // Implement this method for wave 3
+  let totalScore = 0;
+  let flattenScoreChart = {}
+
+  // all object keys are strings, need to convert
+  for (const [score, letters] of Object.entries(SCORECHART)){
+    for (const letter of letters){
+      flattenScoreChart[letter] = Number(score);
+    };
+  };
+
+  for (let letter of word.toUpperCase()){
+    totalScore += flattenScoreChart[letter];
+  }
+
+  if (word.length >= 7){
+    totalScore += 8;
+  }
+
+  return totalScore;
 };
 
 export const highestScoreFrom = (words) => {
   // Implement this method for wave 4
+  let highestScoreWord = '';
+  let maxScore = 0;
+
+  for (let word of words){
```

Comment:
It doesn't look like `word` gets reassigned in the loop so we should use `const`.

Nitpick: missing whitespace.

```suggestion
  for (const word of words) {
```